### PR TITLE
graphene_set.cpp warning fixes

### PR DESCRIPTION
**Pull request by gubatron**
- Multiple `Value of type 'x' may not fit into the receyver type 'y'` warnings fixed.
  - In constructor, vector `cheapHashes` unspecified size initialization warning fixed.

### CODE REVIEW

File: src/blockrelay/graphene_set.cpp [hunk 1]
```diff
@@ -56,11 +56,11 @@ CGrapheneSet::CGrapheneSet(size_t _nReceiverUniverseItems,
 
     // Construct Bloom filter
     pSetFilter = new CBloomFilter(
-        nItems, fpr, insecure_rand.rand32(), BLOOM_UPDATE_ALL, true, std::numeric_limits<uint32_t>::max());
+    static_cast<unsigned int>(nItems), fpr, insecure_rand.rand32(), BLOOM_UPDATE_ALL, true, std::numeric_limits<uint32_t>::max());
     LOG(GRAPHENE, "fp rate: %f Num elements in bloom filter: %d\n", fpr, nItems);
 
     // Construct IBLT
-    uint64_t nIbltCells = std::max((int)IBLT_CELL_MINIMUM, (int)ceil(optSymDiff));
+    uint64_t nIbltCells = (uint64_t) std::max((int)IBLT_CELL_MINIMUM, (int)ceil(optSymDiff));
```
Comment (awemany) [inline src/blockrelay/graphene_set.cpp:63]: Hmm, this looks like a roundabout way with too many casts. First casting to `int` and then to `uint64_t`. Might there be a better way to deal with this, just casting the necessary parts to the same size on the RHS? This goes for quite a few of the fixes below as well.

File: src/blockrelay/graphene_set.cpp [hunk 2]
```diff
@@ -86,12 +86,12 @@ CGrapheneSet::CGrapheneSet(size_t _nReceiverUniverseItems,
 
         mapCheapHashes.clear();
 
-        std::vector<uint64_t> cheapHashes;
+        std::vector<uint64_t> cheapHashes(_itemHashes.size());
```
Comment (awemany) [inline src/blockrelay/graphene_set.cpp:89]: Was `clang` warning about this? Seems like some kind of optimizer warning? Just curious, have not seen that as a warning yet. (Also goes for the one below)

File: src/blockrelay/graphene_set.cpp [hunk 3]
```diff
@@ -254,7 +254,7 @@ std::vector<unsigned char> CGrapheneSet::EncodeRank(std::vector<uint64_t> items,
         assert(ceil(log2(item)) <= nBitsPerItem);
 
         for (uint16_t j = 0; j < nBitsPerItem; j++)
-            bits[j + i * nBitsPerItem] = (item >> j) & 1;
+            bits[j + i * nBitsPerItem] = static_cast<bool>((item >> j) & 1);
```
Comment (awemany) [inline src/blockrelay/graphene_set.cpp:257]: I feel this kind of extra code obfuscates more than it helps. Is there a way to fine tune the warning flags of your compiler to suppress this particular type of warning?

**Review by awemany — CHANGES_REQUESTED** [review]

File: src/blockrelay/graphene_set.cpp (hunk 2 above)
Comment (gubatron) [reply src/blockrelay/graphene_set.cpp:89]: ran a static code analyzer and this was a complaint

Comment (awemany) [reply src/blockrelay/graphene_set.cpp:89]: What analyzer? clang sa?

**Pull request closed without merging**